### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by sabival89**

### CODE REVIEW

File: src/accounts/accounts.controller.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import { WithdrawTransactionDto } from 'src/transactions/dto/withdraw-transaction.dto';
+import { SwyftTablesInfo } from 'src/typings/types';
+import { CreateTransactionDto } from '../transactions/dto/create-transaction.dto';
+import { Transaction } from '../transactions/entities/transaction.entity';
+import { AccountsService } from './accounts.service';
+import { CreateAccountDto } from './dto/create-account.dto';
+import { UpdateAccountDto } from './dto/update-account.dto';
+import { Account } from './entities/account.entity';
+import { ValidationPipe } from './pipes/swyft-validation.pipe';
+
```
Comment (cyrusBaker) [inline src/accounts/accounts.controller.ts:23]: For swagger purposes try adding "@ApiTags('Accounts')" 
you will also need to: 
import { ApiTags } from '@nestjs/swagger';

File: src/main.ts [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
 import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
+
+  const SwaggerConfig = new DocumentBuilder()
+    .setTitle('Swyft API')
+    .setDescription('The Swyft banking API')
+    .setVersion('1.0')
+    .addTag('payment')
```
Comment (cyrusBaker) [inline src/main.ts]: If you do the swagger changes in the controllers you can delete this line: 
.addTag('payment')

File: src/transactions/transactions.controller.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { Controller, Get, HttpException } from '@nestjs/common';
+import { SwyftTablesInfo } from 'src/typings/types';
+import { TransactionsService } from './transactions.service';
+
```
Comment (cyrusBaker) [inline src/transactions/transactions.controller.ts:5]: For swagger purposes try adding "@ApiTags('Transactions')" 
you will also need to: 
import { ApiTags } from '@nestjs/swagger';

File: src/accounts/dto/create-account.dto.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+export class BalanceAttributes {
+  @ApiProperty()
+  @IsNumber()
+  @Min(0)
+  readonly amount: number;
+
+  @ApiProperty()
+  @IsString()
+  @Contains('USD')
```
Comment (cyrusBaker) [inline src/accounts/dto/create-account.dto.ts]: @IsEnum(AvailableCurrencies)

File: src/accounts/accounts.controller.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  /**
+   * Handler for account update
+   * @param accountId Account ID of the target account for update
+   * @param updateAccountDto payload
+   * @returns
+   */
+  @Patch(':id')
+  async updateAccount(
+    @Param('id') accountId: string,
```
Comment (cyrusBaker) [inline src/accounts/accounts.controller.ts]: @Param('id', new ParseUUIDPipe()) accountId: string,

add this whenever you have a uuid param

File: src/accounts/dto/create-account.dto.ts (hunk 4 above)
Comment (sabival89) [reply src/accounts/dto/create-account.dto.ts]: How do I make the placeholder value for currency display 'USD' on swagger?

File: src/accounts/accounts.controller.ts (hunk 5 above)
Comment (sabival89) [reply src/accounts/accounts.controller.ts]: Awesome. Is there a difference between calling ParseUUIDPipe and new ParseUUIDPipe(). The ParseUUIDPipe works for me.

File: src/core/decorators/is-table-empty.decorator.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  descriptor.value = async function (...args: any): Promise<any> {
+    return Repository.isTableEmpty(table)
+      .then(() => mainMethod.apply(this, args))
+      .catch(() => {
+        return new BadRequestException('Database is empty');
+      });
+  };
+
+  return descriptor;
+};
```
Comment (cyrusBaker) [inline src/core/decorators/is-table-empty.decorator.ts:25]: Nice!

**Review by cyrusBaker — APPROVED** [review]

**Pull request merged**